test/test_nn.py

```diff
@@ -11910,6 +11910,13 @@ def test_leaky_relu_inplace_overlap(self, device):
         with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
             F.leaky_relu_(x)
 
+    def test_threshold_inplace_overlap(self, device):
+        x = torch.randn((1, 6), device=device).expand((6, 6))
+        with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
+            F.threshold(x, 0.5, 0.5, inplace=True)
+        with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
+            F.threshold_(x, 0.5, 0.5)
```

Comment:
Actually, in-place threshold is one of the cases where we can get BC-breaking behavior, right? The following would fail after this PR:
```
import torch
import torch.nn.functional as F
x = F.threshold(torch.tensor([1., -1]).expand(5, 2), threshold=0, value=0.5, inplace=True)
```
So just like `fill_` I think we should have an exception for in-place threshold